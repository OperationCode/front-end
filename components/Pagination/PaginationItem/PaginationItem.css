.PaginationItem {
  height: 2em;
  width: 2em;
  display: flex;
  justify-content: center;
  align-items: center;
  margin: 5px 0;
  border: 2px solid transparent;
  cursor: pointer;
  font-family: var(--secondaryFontFamily);
}

.PaginationItem > svg {
  fill: currentColor;
  color: var(--primary);
  height: 1.5em;
}

.current {
  border-color: var(--primary);
}

.notClickable {
  cursor: default;
}
