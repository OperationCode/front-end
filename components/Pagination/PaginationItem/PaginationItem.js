import React from 'react';
import Link from 'next/link';
import { bool, node, number, string } from 'prop-types';
import classNames from 'classnames';
import ScreenReaderOnly from 'components/ScreenReaderOnly/ScreenReaderOnly';
import styles from './PaginationItem.module.css';

PaginationItem.propTypes = {
  children: node.isRequired,
  isCurrent: bool,
  pathname: string.isRequired,
  testId: string.isRequired,
  value: number,
};

PaginationItem.defaultProps = {
  isCurrent: false,
  value: undefined,
};

function PaginationItem({ children, isCurrent, pathname, testId, value }) {
  const isClickable = !!value;

  return (
    <li
      className={classNames(styles.PaginationItem, {
        [styles.current]: isCurrent,
        [styles.notClickable]: !isClickable,
      })}
      data-testid={testId}
    >
      {isClickable ? (
        <Link href={pathname} as={pathname.replace('[page]', value)}>
          <a className={styles.unstyledLink}>
            <ScreenReaderOnly>Go to page</ScreenReaderOnly>
            {children}
          </a>
        </Link>
      ) : (
        children
      )}
    </li>
  );
}

export default PaginationItem;
