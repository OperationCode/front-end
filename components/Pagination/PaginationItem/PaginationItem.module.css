.PaginationItem {
  height: 2em;
  width: 2em;
  display: flex;
  justify-content: center;
  align-items: center;
  margin: 5px 0;
  border: 2px solid transparent;
  cursor: pointer;
  font-family: var(--secondaryFontFamily);
}

.PaginationItem > svg {
  fill: currentColor;
  color: var(--primary);
  height: 1.5em;
}

.current {
  border-color: var(--primary);
}

.notClickable {
  opacity: 0.8;
  cursor: default;
}

.unstyledLink {
  display: flex;
  justify-content: center;
  align-items: center;
  border: none;
  appearance: none;
  background-color: inherit;
  font-size: inherit;
  color: inherit;
  outline: none;
  padding: 0;
  margin: 0;
  width: 100%;
  height: 100%;
}

.unstyledLink:focus {
  border: 2px dashed var(--primary);
  outline: none;
}

.unstyledLink:hover {
  cursor: pointer;
}
