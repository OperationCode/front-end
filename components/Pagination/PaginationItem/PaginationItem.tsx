import Link from 'next/link';
import omit from 'lodash/omit';
import classNames from 'classnames';
import styles from './PaginationItem.module.css';
import { ScreenReaderOnly } from '@/components/ScreenReaderOnly/ScreenReaderOnly';

export interface PaginationItemPropsType {
  /**
   * Content to be rendered as the link.
   */
  children: React.ReactNode;
  /**
   * Sets the URL path
   */
  pathname: string;
  /**
   * Sets an id to the base element for testing.
   */
  testId: string;
  /**
   * Sets styles to indicate the current item is selected.
   */
  isCurrent?: boolean;
  query?: Record<string, unknown>;
  value?: number;
}

export function PaginationItem({
  children,
  isCurrent = false,
  pathname,
  query,
  testId,
  value,
}: PaginationItemPropsType) {
  const relevantQueryStringObject = omit(query, ['page']);
  const realURL = {
    pathname: pathname.replace('[page]', String(value)),
    query: relevantQueryStringObject,
  };

  const isClickable = !!value;

  return (
    <li
      className={classNames(styles.PaginationItem, {
        [styles.current]: isCurrent,
        [styles.notClickable]: !isClickable,
      })}
      data-testid={testId}
    >
      {isClickable ? (
        <Link href={{ pathname, query }} as={realURL} className={styles.unstyledLink}>
          <ScreenReaderOnly>Go to page</ScreenReaderOnly>
          {children}
        </Link>
      ) : (
        children
      )}
    </li>
  );
}
