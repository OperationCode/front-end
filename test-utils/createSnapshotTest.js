/* eslint-env jest */
import renderer from 'react-test-renderer';

/**
 * Used to create fully-rendered snapshot test.
 * Useful for reusable component smoke tests.
 *
 * @export
 * @param {*} Component
 */
export default Component => {
  const tree = renderer.create(Component).toJSON();
  expect(tree).toMatchSnapshot();
};
