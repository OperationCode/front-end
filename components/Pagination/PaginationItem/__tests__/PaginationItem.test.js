import React from 'react';
import { cleanup, render } from '@testing-library/react';
import createSnapshotTest from 'test-utils/createSnapshotTest';

import PaginationItem from '../PaginationItem';

describe('PaginationItem', () => {
  const requiredProps = {
    children: '1',
    pathname: '/resources/[page]',
    query: {},
    testId: 'test',
  };

  afterEach(cleanup);

  it('renders with required props', () => {
    createSnapshotTest(<PaginationItem {...requiredProps} />);
  });

  it('renders correctly as a current item with value', () => {
    createSnapshotTest(<PaginationItem {...requiredProps} value={1} isCurrent />);
  });

  it('should be marked as current if current', () => {
    const { queryByTestId } = render(<PaginationItem {...requiredProps} value={1} isCurrent />);

    expect(queryByTestId(requiredProps.testId)).toHaveClass('current');
  });

  it('is not clickable when value is undefined', () => {
    const { queryByTestId } = render(<PaginationItem {...requiredProps} />);

    expect(queryByTestId(requiredProps.testId)).toHaveClass('notClickable');
  });
});
