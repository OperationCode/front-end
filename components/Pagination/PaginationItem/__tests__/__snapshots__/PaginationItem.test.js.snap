// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PaginationItem renders correctly as a current item with value 1`] = `
<li
  className="PaginationItem current"
  data-testid="test"
>
  <a
    className="unstyledLink"
    href="/resources/1"
    onClick={[Function]}
    onMouseEnter={[Function]}
  >
    <span
      className="ScreenReaderOnly"
      data-testid="SCREEN_READER_ONLY"
    >
      Go to page
    </span>
    1
  </a>
</li>
`;

exports[`PaginationItem renders with required props 1`] = `
<li
  className="PaginationItem notClickable"
  data-testid="test"
>
  1
</li>
`;
